### PR TITLE
smack: move helper function to smack-util.c

### PR DESCRIPTION
**Pull request by keszybz**
The function was in basic/fileio.c, but it's more appropriate to
keep it out of src/basic.

Fixes compilation with -Dsmack-run-label= set.

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
lgtm

**Review by bluca — APPROVED** [review]
lgtm

**Comment by keszybz** [comment]
Please take a look again, I added one more commit to make the configuration self-consistent.

**Comment by keszybz** [comment]
I think I screwed this up a bit when converting to meson…

**Comment by yuwata** [comment]
LGTM.

File: src/shared/smack-util.c [hunk 1]
```diff
@@ -284,3 +284,16 @@ int mac_smack_copy(const char *dest, const char *src) {
...
+        int r = 0;
+        if (rename(from, to) < 0)
+                return -errno;
+
+#if HAVE_SMACK_RUN_LABEL
+        r = mac_smack_apply(to, SMACK_ATTR_ACCESS, SMACK_FLOOR_LABEL);
+        if (r < 0)
+                return r;
+#endif
+        return r;
```
Comment (poettering) [inline src/shared/smack-util.c:298]: the function is so weird. i.e. all those dealings with `r` that are totally unnecessary...

**Pull request merged**